cargo-dist/src/lib.rs

```diff
@@ -438,6 +559,11 @@ fn build_cargo_target(dist_graph: &DistGraph, target: &CargoBuildStep) -> Result
             eprintln!(" --package={})", package);
         }
     }
+    // If we generated any extra environment variables to
+    // inject into the environment, apply them now.
+    for (key, value) in desired_extra_env {
+        command.env(key, value);
+    }
```

Comment:
Checking you're aware of https://doc.rust-lang.org/std/process/struct.Command.html#method.envs ?